pkg/softimpersonation/softimpersonation.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package softimpersonation
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+
+	authenticationv1 "k8s.io/api/authentication/v1"
+	kuser "k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	softImpersonationHeader = "X-Kcp-Internal-Soft-Impersonation"
```

Comment:
who consumes it? (it's private now)

Reply:
It is consumed through the public `UserInfoFromRequestHeader` function.